.github/workflows/server.yml

```diff
@@ -0,0 +1,16 @@
+on:
+  push:
+
+jobs:
+  server:
+    runs-on: ubuntu-latest
+    container: cimg/android:2022.12.1
```

Comment:
Who created this image? Can we get an image with the necessary to build both the server and the client? (sorry, I'm a newbie in this domain)

Reply:
These images are generated and maintained by Circle CI, a provider of CI tooling & infrastructure.  I had a look at the container images which can be used for building Android apps, and this seemed to be one of the most maintained ones.  The source repo is here: https://github.com/CircleCI-Public/cimg-android.

That's a long way to say I think they are fairly trustworthy.  I tried to start from ubuntu:jammy and then install ADK from the Ubuntu package repositories, but the ADK that ships with Ubuntu is too old.  The other option is to start from a vanilla ubuntu:jammy image (or the vanilla image for your favorite distro) and manually install the ADK.  Or not run in a container at all, but directly on the GitHub actions VM, which ship with the ADK as well: https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2204-Readme.md#android.

In general, I think you're better of using containers because it shields you from any changes on the underlying build machines.

For the app, I would recommend running the build for the app in a separate job in a separate container, as the build dependencies are quite different: the server requires the ADK, the app itself requires headers for libav*, libusb, mingw and a bunch of other dependencies.

I originally had a job which would build the client as well, but then decided against it as I didn't have the time to fully complete that work (done is better than perfect, I guess).

Reply:
My dev container dockerfile is also based on the cimg/android image, just only contains the essential components for building Scrcpy: https://github.com/yume-chan/scrcpy-devcontainer/blob/9a7b20391574d25617241d07b75457c9f22dd1a2/.devcontainer/Dockerfile#L30-L51